source/sessions/snapshot-sessions.rst

```diff
@@ -0,0 +1,289 @@
+
+Version 5.0 of the server introduces support for read concern level "snapshot" (non-speculative)
+for read commands outside of transactions, including on secondaries.
+This spec builds upon the `Sessions Specification <../driver-sessions.rst>`_ to define how an application
+requests "snapshot" level read concern and how a driver interacts with the server
+to implement snapshot reads.
+
+Definitions
+===========
+
+META
+----
+
+The keywords “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”,
+“SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be
+interpreted as described in `RFC 2119 <https://www.ietf.org/rfc/rfc2119.txt>`_.
+
+Terms
+-----
+
+ClientSession
+    The driver object representing a client session and the operations that can be
+    performed on it.
+
+MongoClient
+    The root object of a driver's API. MAY be named differently in some drivers.
+
+MongoCollection
+    The driver object representing a collection and the operations that can be
+    performed on it. MAY be named differently in some drivers.
+
+MongoDatabase
+    The driver object representing a database and the operations that can be
+    performed on it. MAY be named differently in some drivers.
+
+ServerSession
+    The driver object representing a server session.
+
+Session
+    A session is an abstract concept that represents a set of sequential
+    operations executed by an application that are related in some way. This
+    specification defines how sessions are used to implement snapshot reads.
+
+Snapshot reads
+    Reads with read concern level ``snapshot`` that occur outside of transactions on
+    both the primary and secondary nodes, including in sharded clusters.
+    Snapshots reads are majority committed reads.
+
+Snapshot timestamp
+    Snapshot timestamp, representing timestamp of the first read (find/aggregate/distinct operation) in the session.
+    The server creates a cursor in response to a snapshot find/aggregate command and 
+    reports ``atClusterTime`` field in the cursor response. For distinct command server adds ``atClusterTime`` field to the distinct response object.
+    ``atClusterTime`` field represents the timestamp of the read and is guaranteed to be majority committed.
+
+Specification
+=============
+
+An application requests snapshot reads by creating a ``ClientSession``
+with options that specify that snapshot reads are desired. An
+application then passes the session as an argument to methods in the
+``MongoDatabase`` and ``MongoCollection`` classes. Read operations (find/aggregate/distinct) performed against
+that session will be read from the same snapshot.
+
+High level summary of the API changes for snapshot reads
+========================================================
+
+Snapshot reads are built on top of client sessions.
+
+Applications will start a new client session for snapshot reads like
+this:
+
+.. code:: typescript
+
+    options = new SessionOptions(isSnapshot = true);
```

Comment:
Can we rename `isSnapshot` to `snapshot`? Including `is` in the name doesn't add any meaning, it's just a filler word. It's also inconsistent with other boolean options in the driver api. For example `causalConsistency` isn't called `isCausalConsistency` or `isCausallyConsistent`.

Reply:
Good point re. consistency. Renamed.